### PR TITLE
UI: Add snap-to-center in preview

### PR DESCRIPTION
**Pull request by Fishrock123**
Adds center (horizontal & vertical) snapping for sources in the preview pane.

It's possible my math isn't quite correct in all cases.

~~Ideally, I'd like to add options to adjust snapping so that people can enable/disable/adjust the clamp size, but that is a bit more complex. :P~~
Edit: done!

### CODE REVIEW

**Comment by ONSBalder** [comment]
Can one of the admins verify this patch?

**Comment by Fishrock123** [comment]
Updated to add settings as @dodgepong suggested I should go ahead with IRC.

**Comment by Fishrock123** [comment]
As a note, the `snapDistance` always sets to 2.0 for some reason once it is adjusted.

File: obs/forms/OBSBasicSettings.ui [hunk 1]
```diff
@@ -6,8 +6,8 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>896</width>
-    <height>667</height>
+    <width>939</width>
+    <height>674</height>
```
Comment (Fishrock123) [inline obs/forms/OBSBasicSettings.ui]: I edited this with qt designer, should these not have been changed?

**Pull request closed without merging**